### ISSUE TITLE
Null reference in VersionControl after modifying solution file outside IDE

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#7313]
Triggered this with the designer's solution file by doing:

   touch Xamarin.FormsDesigner.MonoDevelop.sln

The solution is then reloaded in the IDE. See some null reference exceptions and an object disposed exception.

Release ID: 800002696
Git revision: ea5afd016ef4420b8d93ad9e47269580f0996b52
Build date: 2019-03-06 06:58:54+00
Build branch: master
Xamarin extensions: 1699e68b1cd99f87ad034102c83872ef162916ca

```
ERROR [2019-03-06 13:54:09Z]: System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.VersionControl.Repository.GetDirectoryVersionInfo (MonoDevelop.Core.FilePath localDirectory, System.Boolean getRemoteStatus, System.Boolean recursive) [0x00056] in /Users/vsts/agent/2.148.0/work/1/s/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/Repository.cs:308 
  at MonoDevelop.VersionControl.VersionControlNodeExtension.BuildNode (MonoDevelop.Ide.Gui.Components.ITreeBuilder builder, System.Object dataObject, MonoDevelop.Ide.Gui.Components.NodeInfo nodeInfo) [0x00030] in /Users/vsts/agent/2.148.0/work/1/s/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/VersionControlNodeExtension.cs:61 
  at MonoDevelop.Ide.Gui.Components.ExtensibleTreeView+TreeBuilder.GetNodeInfo (MonoDevelop.Ide.Gui.Components.NodeInfo nodeInfo, MonoDevelop.Ide.Gui.Components.ExtensibleTreeView tree, MonoDevelop.Ide.Gui.Components.ITreeBuilder tb, MonoDevelop.Ide.Gui.Components.NodeBuilder[] chain, System.Object dataObject) [0x00011] in /Users/vsts/agent/2.148.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/TreeBuilder.cs:304 
ERROR [2019-03-06 13:54:09Z]: System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'GitRepository'.
  at MonoDevelop.VersionControl.Git.GitRepository.EnsureInitialized () [0x00032] in /Users/vsts/agent/2.148.0/work/1/s/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:445 
  at MonoDevelop.VersionControl.Git.GitRepository.RunOperation[T] (System.Func`1[TResult] action, System.Boolean hasUICallbacks) [0x00000] in /Users/vsts/agent/2.148.0/work/1/s/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:477 
  at MonoDevelop.VersionControl.Git.GitRepository.GetCurrentBranch () [0x00000] in /Users/vsts/agent/2.148.0/work/1/s/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:1676 
  at MonoDevelop.VersionControl.Git.GitNodeBuilderExtension.BuildNode (MonoDevelop.Ide.Gui.Components.ITreeBuilder treeBuilder, System.Object dataObject, MonoDevelop.Ide.Gui.Components.NodeInfo nodeInfo) [0x00032] in /Users/vsts/agent/2.148.0/work/1/s/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitNodeBuilderExtension.cs:67 
  at MonoDevelop.Ide.Gui.Components.ExtensibleTreeView+TreeBuilder.GetNodeInfo (MonoDevelop.Ide.Gui.Components.NodeInfo nodeInfo, MonoDevelop.Ide.Gui.Components.ExtensibleTreeView tree, MonoDevelop.Ide.Gui.Components.ITreeBuilder tb, MonoDevelop.Ide.Gui.Components.NodeBuilder[] chain, System.Object dataObject) [0x00011] in /Users/vsts/agent/2.148.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/TreeBuilder.cs:304 
```

> VS bug [#812517](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/812517)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-9] Comment by **xamarinc**



Fixed in version 8.0.0.2798 (release-8.0)



Pull Request #321 merged by: Vsevolod Kukol  
 Author: xamarin  
 Commit: 337bfdca90da1359caf8eb746c70332fad5d7e55 (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-11] Comment by **xamarinc**



Fixed in version 8.1.0.1701 (master-vnext)



Pull Request #321 merged by: Vsevolod Kukol  
 Author: xamarin  
 Commit: 337bfdca90da1359caf8eb746c70332fad5d7e55 (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-20] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #321 merged by: Vsevolod Kukol  
 Author: xamarin  
 Commit: 337bfdca90da1359caf8eb746c70332fad5d7e55 (xamarin/monodevelop)
